Monika After Story/game/zz_poems.rpy

```diff
@@ -0,0 +1,284 @@
+
+        OUT:
+            A list of poems based on the specifications above
+        """
+
+        #If we only want unseen, do this
+        if unseen:
+            return [
+                poem
+                for poem in poem_map.itervalues()
+                if not poem.is_seen() and poem.category == category
+            ]
+
+        #Otherwise we just get all
+        return [
+            poem
+            for poem in poem_map.itervalues()
+            if poem.category == category
+        ]
+
+    def getSeenPoems():
+        """
+        Returns a list of all seen poems ordered by category
+        """
+        return sorted([
+            poem
+            for poem in poem_map.itervalues()
+            if poem.is_seen()
+        ], key=poem_sort_key)
+
+    def getUnseenPoems():
+        """
+        Returns a list of all unseen poems ordered by category
+        """
+        return sorted([
+            poem
+            for poem in poem_map.itervalues()
+            if not poem.is_seen()
+        ], key=poem_sort_key)
+
+    def getPoem(poem_id):
+        """
+        Gets a poem by id
+
+        IN:
+            poem_id - poem id of the poem to get
+
+        OUT:
+            MASPoem if there's a poem with the id
+            None if no poem with the id exists
+        """
+        return poem_map.get(poem_id, None)
+
+    def getSeenPoemsMenu():
+        """
+        Gets a list of seen poems in scrollable menu format
+
+        OUT:
+            A list of seen poems in the format for a mas gen scrollable menu
+        """
+        return [
+            (poem.prompt, poem, False, False)
+            for poem in poem_map.itervalues()
+            if poem.is_seen()
+        ]
+
+
+init 10 python:
+    class MASPoem:
+        def __init__(
+            self,
+            poem_id,
+            category,
+            prompt,
+            title="",
+            text="",
+            author="monika"
+        ):
+            """
+            MASPoem constructor
+
+            Similar to the Poem class from DDLC, but excludes the yuri variables and adds a poem id property.
+
+
+            poem_id:
+                identifier for the poem. (NOTE: Must be unique)
+
+            category:
+                category for the poem is under (So we can get poems by category)
+
+            prompt:
+                prompt for this poem (So it can be viewed by a scrollable menu)
+
+            title:
+                poem title (supports renpy substitution)
+
+            text:
+                poem contents
+
+            author:
+                poem author (Default: monika)
+            """
+            if poem_id in store.mas_poems.poem_map:
+                raise Exception ("poem_id {0} already exists in the poem map.".format(poem_id))
+
+            self.poem_id=poem_id
+            self.category=category
+            self.prompt=prompt
+            self.title=renpy.substitute(title)
+            self.text=renpy.substitute(text)
```

Comment:
using substitute here might not work as intended since poem object creation could be before desired variables are set. better to save the raw text and change `mas_generic_poem` screen to sub text before showing it.